### ISSUE TITLE
+polite

### ISSUE THREAD

**Issue by awfulcooking** [ircv3/ircv3-ideas#106]
Indicates the message is not pressing / doesn't wish to distract (some of) the mentioned nicks.

Up to the recipient if this does anything.

```irc
@+polite=jwheare :A!staff@irccloud.com PRIVMSG #irccloud :B: yeah that'd be good for jwheare to take a look at in the morning

:Foo PRIVMSG #game !scores
@+polite=Wodda,Score,Jeff :Bot PRIVMSG #game :Foo: top scores | Wodda 100 | Score 90 | Jeff 80

@+polite=emersion :val PRIVMSG #project :i think emersion tried that at one point, then Amelie picked it up.
```

**Comment by emersion** [comment]
A hack I use to avoid highlighting users is to insert a zero-width space (U+200B) after the first character.

**Comment by awfulcooking** [comment]
> A hack I use to avoid highlighting users is to insert a zero-width space (U+200B) after the first character.

Yes, that's excellent in some ways, but then I worry that someone seeing their nick not be underlined, highlighted or hoverable without explanation would be even more distracting..

Edit: imagine they copy it into a search, etc

**Comment by jesopo** [comment]
> A hack I use to avoid highlighting users is to insert a zero-width space (U+200B) after the first character.

my bot used to do this until i realised it was both rendering as an actual space in some clients and breaking character width calculations in some terminals

**Comment by emersion** [comment]
> someone seeing their nick not be underlined, highlighted or hoverable without explanation would be even more distracting

Right.

> rendering as an actual space in some clients and breaking character width calculations in some terminals

These seem like client bugs, ie. not something that needs a new spec.

**Comment by progval** [comment]
Nitpick: the separator should be `\s` instead of comma because spec-wise, commas are allowed in nicks

EDIT (2023-08-25): actually they aren't, because commas are already used as a separator in KICK (and PRIVMSG)

**Comment by jesopo** [comment]
> These seem like client bugs, ie. not something that needs a new spec.

one of these cases is actually a bug in Mac OS' terminal, which is out of scope for us to fix. i think making highlights an explicit act is a good idea, though i'm not sure about doing it by `+polite`

**Comment by awfulcooking** [comment]
The experience I pictured was your nick still being pilled or linked, and maybe appearing still in your highlight drawer, just not with a notification

**Comment by awfulcooking** [comment]
> These seem like client bugs, ie. not something that needs a new spec.

It not being underlined / clickable would still be a significant UX bug, and not being fixable without "e\u200Bmersion" becoming the spec

**Comment by emersion** [comment]
> one of these cases is actually a bug in Mac OS' terminal, which is out of scope for us to fix

If a client wanted to workaround this, they could strip out any U+200B from the message text after looking for highlights.

> i think making highlights an explicit act is a good idea, though i'm not sure about doing it by +polite

Yup, same here.

An alternative would be to make highlights explicit by requiring a tag or a new formatting character to be present to represent highlights. Backwards compat needs to be addressed with this approach though.

**Comment by nektro** [comment]
there could be a capability for mention references to coordinate the backwards compat. enabled it could be like:

`nektro` is a polite mention

`<@nektro>` is an explicit mention

for clients that dont support the capability, the server can convert the latter back into the former

**Comment by awfulcooking** [comment]
That's an interesting idea @nektro. In that paradigm, would you expect the server to be polite-by-default?

`CAP REQ polite-mentions`

**Comment by nektro** [comment]
thanks, hmm I was more thinking about a UI that would parse the message and send the right one based on whether or not the user wanted it to be polite. if we wanted to optimize for hand-written messages to some degree i think keeping do-ping by default is the way to go. maybe `nektro` vs `!nektro!`

although, the `!nektro!` almost looks like a "super-ping" but im hesitant to use `<nektro>` for the non-ping version since ive seen this syntax used to reply to folks a lot before

having the ping-ness in-band is ideal i think but perhaps having it in the message metadata would indeed be better

having it in-band makes messages easier to write but the backwards compatibility easier or harder depending on what you're optimizing for
having it in the message meta puts the onus on clients to implement a UI to disable the ping-ness of a message, something like twitter's reply options

![image](https://github.com/ircv3/ircv3-ideas/assets/5464072/5b5eafe7-3532-4e89-951e-e731c6653b74)

**Comment by acidvegas** [comment]
L O L

**Comment by awknode** [comment]
hahahahahaha

**Comment by kiwijc** [comment]
Lmao
